common/src/main/java/com/wynntils/screens/content/WynntilsCaveScreen.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.content;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.mc.event.MenuEvent;
+import com.wynntils.models.content.CaveInfo;
+import com.wynntils.models.content.event.ContentTrackerUpdatedEvent;
+import com.wynntils.models.content.event.ContentUpdatedEvent;
+import com.wynntils.models.content.type.ContentSortOrder;
+import com.wynntils.models.content.type.ContentType;
+import com.wynntils.screens.base.WynntilsListScreen;
+import com.wynntils.screens.base.widgets.BackButton;
+import com.wynntils.screens.base.widgets.PageSelectorButton;
+import com.wynntils.screens.base.widgets.ReloadButton;
+import com.wynntils.screens.base.widgets.SortOrderWidget;
+import com.wynntils.screens.base.widgets.SortableContentScreen;
+import com.wynntils.screens.content.widgets.CaveButton;
+import com.wynntils.screens.wynntilsmenu.WynntilsMenuScreen;
+import com.wynntils.utils.StringUtils;
+import com.wynntils.utils.colors.CommonColors;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.render.FontRenderer;
+import com.wynntils.utils.render.Texture;
+import com.wynntils.utils.render.type.HorizontalAlignment;
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import net.minecraft.client.resources.language.I18n;
+import net.minecraft.network.chat.Component;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public final class WynntilsCaveScreen extends WynntilsListScreen<CaveInfo, CaveButton>
+        implements SortableContentScreen {
+    private ContentSortOrder contentSortOrder = ContentSortOrder.LEVEL;
+    private CaveInfo trackingRequested = null;
+
+    private WynntilsCaveScreen() {
+        super(Component.translatable("screens.wynntils.wynntilsCaveBook.name"));
+
+        // Only register this once
+        WynntilsMod.registerEventListener(this);
+    }
+
+    @SubscribeEvent
+    public void onCaveUpdate(ContentUpdatedEvent event) {
+        if (event.getContentType() == ContentType.CAVE && McUtils.mc().screen == this) {
+            this.reloadElements();
+        }
+    }
+
+    @SubscribeEvent
+    public void onTrackedContentUpdate(ContentTrackerUpdatedEvent event) {
+        // Reload so we have the proper order
+        if (McUtils.mc().screen == this) {
+            this.reloadElements();
+        }
+    }
+
+    // FIXME: We only need this hack to stop the screen from closing when tracking Caves.
+    //        Adding a proper way to add quests with scripted container queries would mean this can get removed.
```

Comment:
Not sure I understand this..?

Reply:
This is copied from the quest book, not a new invention. When you track a quest/cave, it will close the screen (because wynn closes the book after tracking).

Reply:
But the comment? "Adding a proper way to add quests with scripted container queries would mean this can get removed." -- what is missing from the scripted container queries?

Reply:
We don't care if the screen get's closed by the server, we just end our query, send a packet to close the screen and go on. We need to add a query step, something like "expectContainerClose".

Reply:
Ah, now I see. When the server closes the container after selecting a content item for tracking, we still send a request to close the screen. https://github.com/Wynntils/Artemis/issues/1808